sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 9/11/2022.
+ */
+
+package com.adyen.checkout.sessions.provider
+
+import com.adyen.checkout.components.base.Configuration
+import com.adyen.checkout.sessions.CheckoutSession
+import com.adyen.checkout.sessions.model.SessionModel
+import kotlinx.coroutines.CoroutineScope
+
+// TODO docs
+object CheckoutSessionProvider {
+    // TODO docs
+    suspend fun createSession(
+        sessionModel: SessionModel,
+        configuration: Configuration,
+    ): Result<CheckoutSession> {
```

Comment:
Not totally sure if we should expose `Result` publicly 🤔 Some people really dislike it

Reply:
Yeah i had a sealed class first but I removed it for some reason, will add it back 👍